pkg/rkeworker/docker.go

```diff
@@ -107,7 +111,10 @@ func changed(ctx context.Context, c *client.Client, p v3.Process, container type
 	if err != nil {
 		return false, err
 	}
-
+	imageInspect, _, err := c.ImageInspectWithRaw(ctx, inspect.Config.Image)
```

Comment:
You want to use inspect.Image, not Config.Image.  Config.Image is the friendly name that may or may not exist or has change (because of the tag).  inpsect.Image should be the hash that will not change.